core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt

```diff
@@ -78,37 +81,52 @@ class SearchActivity : BaseActivity() {
     )
   }
 
-  override fun injection(coreComponent: CoreComponent) {
-    activityComponent.inject(this)
+  override fun inject(baseActivity: BaseActivity) {
+    baseActivity.cachedComponent.inject(this)
   }
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_search)
-    setSupportActionBar(toolbar)
-    supportActionBar!!.setHomeAsUpIndicator(R.drawable.ic_action_back)
-    supportActionBar!!.setHomeButtonEnabled(true)
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val root = inflater.inflate(R.layout.fragment_search, container, false)
+    setHasOptionsMenu(true)
+    return root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    setupToolbar(view)
     search_list.run {
       adapter = searchAdapter
       layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
       setHasFixedSize(true)
     }
-    compositeDisposable.add(searchViewModel.effects.subscribe { it.invokeWith(this) })
+    compositeDisposable.add(searchViewModel.effects.subscribe {
+      it.invokeWith(requireActivity() as CoreMainActivity)
```

Comment:
I am also not against an extension on Fragment `coreMainActivity` feels like we may repeat this line a bit